pages/jobs.tsx

```diff
@@ -0,0 +1,71 @@
+// import Link from 'next/link'
+import { useState, useEffect } from 'react'
+import { GetStaticProps } from 'next'
+import { siteConfig } from '@/const/meta'
+import Content from 'components/Layout/Content'
+import { Section, Title, SubTitle, TitleSmall, LinkContainer, LoadingText } from 'const/styles/pages/content'
+import { ExternalLink } from '@/const/styles/global'
+import SVG from 'react-inlinesvg'
+
+async function getJobs() {
+  const { api, deptID } = siteConfig.greenhouse
+  const getJobs = await fetch(api)
+  const { jobs } = await getJobs.json()
+  const jobsData = []
+  await jobs.map((job) => {
+    const dept = job.departments[0]
+    if (dept) {
+      dept.id === deptID ? jobsData.push(job) : null
+    }
+  })
+
+  return jobsData
+}
+
+export default function Jobs({ jobsData }) {
+  const discordURL = siteConfig.social.discord.url
+  const [jobs, setJobs] = useState(jobsData)
+  const [isLoading, setLoading] = useState(false)
+
+  // Fetch the latest jobs to possibly override static rendered (getStaticProps) SSR jobs
+  useEffect(() => {
+    setLoading(true)
+    getJobs()
+      .then((data) => {
+        setJobs(data)
+        setLoading(false)
+      })
+  }, [])
+
+  return (
+    <Content>
+      <Section>
+        <Title>Want to build the future of decentralized trading?</Title>
+        <SubTitle>We are an ambitious, fast growing and international team working at the forefront of DeFi. We believe that we can make markets both more efficient and fair, by building the ultimate batch auction settlement layer across EVM compatible blockchains.</SubTitle>
+      </Section>
+
+      <TitleSmall>Who we are looking for:</TitleSmall>
+
+      {isLoading && <LoadingText>Loading jobs...</LoadingText>}
+      {!isLoading && jobs.length > 0 && jobs.map(({ title, absolute_url, location, internal_job_id }) =>
+        <LinkContainer key={internal_job_id} href={absolute_url} target="_blank" rel="noopener nofollow noreferrer">
+          <b>{title}</b>
+          <i>{location.name}</i>
+          <SVG src="images/icons/arrowRight.svg" />
+        </LinkContainer>
+      )}
+      {/* {!isLoading && jobs.length < 1 && <p>There are currently no open positions.</p>} */}
+
+      <p>No open positions or can&apos;t find the position you&apos;re looking for? <ExternalLink href={discordURL} target="_blank" rel="noopener nofollow">Get in touch</ExternalLink></p>
+
+    </Content>
+  )
+}
+
+export const getStaticProps: GetStaticProps = async () => {
```

Comment:
should you use revalidate?

i.e. `revalidate: 3600` so its revalidated every hour?

Reply:
That is in relation to this https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration

I wasn't sure if in this case this would be helpful. This is how they mention it working in this demo: https://reactions-demo.vercel.app/

Which makes sense for Tweets or other more dynamic content, but perhaps not for these jobs positions? My thinking was the user only needs to update once.

Simultaneously I could see this ISR (revalidate) being helpful given the latest fetched jobs (client side) are cached for the user?

But could be overkill as well. Let me know your thoughts.

Reply:
> Which makes sense for Tweets or other more dynamic content, but perhaps not for these jobs positions? My thinking was the user only needs to update once.

well, the positions are updated. I think the `revalidate: 3600` is fair cache policy. Note that users will always see the latest generated content. This just mean that after 1h, the next visit will trigger a generation (although the user still see the stale data). So no performance issue, and we don't need to redeploy if there's a new job offer

Reply:
I will add it, though I'm still not sure if this isn't SSR vs. simply being a cache policy. Reading further down on https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration it seems to require setting up a `secret` for an API of sorts. Something to review post merge of this PR.

Reply:
I would think that this is just for manual revalidation, which is nice to have too in case we really want to invalidate the cache. However, i think it would just work without it good enough